core/src/main/java/hudson/Functions.java

```diff
@@ -1771,7 +1770,15 @@ public boolean hyperlinkMatchesCurrentPage(String href) throws UnsupportedEncodi
         if(Jenkins.getInstanceOrNull()==null)  return Collections.emptyList();
         return PageDecorator.all();
     }
-    
+    /**
+     * Gets all the {@link LoginPageDecorator}s.
+     */
+    public static LoginPageDecorator getLoginPageDecorator() {
+        // this method may be called to render start up errors, at which point Hudson doesn't exist yet. see HUDSON-3608
```

Comment:
Really? AFAICT this is going to be called only by the login screen, so Jenkins is granted to be fully up and running.

I think the `getInstanceOrNull` check can be removed.